plugin/trino-redshift/src/main/java/io/trino/plugin/redshift/RedshiftUnloadSplitManager.java

```diff
@@ -0,0 +1,195 @@
+    private final String unloadLocation;
+    private final String unloadAuthorization;
+    private final ExecutorService executor;
+
+    @Inject
+    public RedshiftUnloadSplitManager(
+            JdbcClient jdbcClient,
+            QueryBuilder queryBuilder,
+            RemoteQueryModifier queryModifier,
+            JdbcSplitManager jdbcSplitManager,
+            RedshiftConfig redshiftConfig,
+            @ForRecordCursor ExecutorService executor)
+    {
+        this.jdbcClient = requireNonNull(jdbcClient, "jdbcClient is null");
+        this.queryBuilder = requireNonNull(queryBuilder, "queryBuilder is null");
+        this.queryModifier = requireNonNull(queryModifier, "queryModifier is null");
+        this.jdbcSplitManager = requireNonNull(jdbcSplitManager, "jdbcSplitManager is null");
+
+        this.unloadLocation = redshiftConfig.getUnloadLocation()
+                .map(location -> location.replaceAll("/$", "") + "/")
+                .orElse(null);
+        if (redshiftConfig.getIamRole().isPresent()) {
+            this.unloadAuthorization = "IAM_ROLE '%s'".formatted(redshiftConfig.getIamRole().get());
+        }
+        else {
+            this.unloadAuthorization = "";
+        }
+        this.executor = requireNonNull(executor, "executor is null");
+    }
+
+    @Override
+    public ConnectorSplitSource getSplits(ConnectorTransactionHandle transaction, ConnectorSession session, ConnectorTableHandle table, DynamicFilter dynamicFilter, Constraint constraint)
+    {
+        if (table instanceof JdbcProcedureHandle) {
+            return jdbcSplitManager.getSplits(transaction, session, table, dynamicFilter, constraint);
+        }
+        ConnectorSplitSource fallbackSplitSource = new FixedSplitSource(new JdbcSplit(Optional.empty()));
+        if (!useUnload(session)) {
+            return fallbackSplitSource;
+        }
+        JdbcTableHandle jdbcTableHandle = dynamicFilteringEnabled(session) ? ((JdbcTableHandle) table).intersectedWithConstraint(dynamicFilter.getCurrentPredicate()) : (JdbcTableHandle) table;
+        List<JdbcColumnHandle> columns = jdbcTableHandle.getColumns()
+                .orElseGet(() -> jdbcClient.getColumns(
+                        session,
+                        jdbcTableHandle.getRequiredNamedRelation().getSchemaTableName(),
+                        jdbcTableHandle.getRequiredNamedRelation().getRemoteTableName()));
+
+        if (!isUnloadSupported(((JdbcTableHandle) table), columns)) {
+            log.debug("Unload query contains unsupported characters. Falling back to using JDBC");
+            return fallbackSplitSource;
+        }
+
+        Connection connection;
+        PreparedStatement statement;
+        try {
+            connection = jdbcClient.getConnection(session);
+            String redshiftSelectSql = getRedshiftSelectSql(session, connection, jdbcTableHandle, columns);
+            if (redshiftSelectSql.contains("\\b")) {
```

Comment:
Worth leaving code comment what's the meaning of `\\b`.

Reply:
added